docs/src/dplyr.md

```diff
@@ -0,0 +1,293 @@
+
+```@example 1
+@chain msleep begin 
+  @select :name :sleep_total
+  @rsubset :sleep_total > 16
+end
+```
+
+You will soon see how useful the `@chain` macro is when we start to combine many functions.  
+
+# Back To dplyr Verbs In Action
+
+Now that you know about the `@chain` macro, we will use it throughout the rest of this tutorial. 
+
+## Arrange Or Re-order Rows Using `@orderby`
+
+To arrange (or re-order) rows by a particular column, such as the taxonomic order, list the name of the column you want to arrange the rows by:
+
+```@example 1
+@orderby msleep :order
+```
+
+Now we will select three columns from msleep, arrange the rows by the taxonomic order and then arrange the rows by sleep\_total. Finally, keep the first 10 rows of the data frame.
+
+```@example 1
+@chain msleep begin 
+    @select :name :order :sleep_total
+    @orderby :order :sleep_total
+    first(10)
+end
+```
+
+Same as above, except here we filter the rows for mammals that sleep for 16 or more hours, instead of showing the head of the final data frame:
+
+```@example 1
+@chain msleep begin 
+    @select :name :order :sleep_total
+    @orderby :order :sleep_total 
+    @rsubset :sleep_total > 16
+end
+```
+
+Something slightly more complicated: same as above, except arrange the rows in the `:sleep_total` column in a descending order. For this, use the function `sortperm` with the keyword argument `rev=true`.
+
+```@example 1
+@chain msleep begin 
+    @select :name :order :sleep_total
+    @orderby begin 
+        :order 
+        sortperm(:sleep_total, rev=true)
+    end 
+    @rsubset :sleep_total >= 16
+end
+```
+
+## Create New Columns Using `@transform` and `@rtransform`
+
+The `@transform` macro will add new columns to the data frame. Like with other macros, use `@rtransform` to operate row-wise. Create a new column called `:rem_proportion`, which is the ratio of rem sleep to total amount of sleep. 
+
+```@example 1
+@rtransform msleep :rem_proportion = :sleep_rem / :sleep_total
+```
+
+You can many new columns using `@transform` by placing multiple operations in a block.
+
+```@example 1
+@rtransform msleep begin 
+    :rem_proportion = :sleep_rem / :sleep_total 
+    :bodywt_grams = :bodywt * 1000
+end
+```
+
+## Create summaries of the data frame using `@combine`
+
+The `@combine` macro will create summary statistics for a given column in the data frame, such as finding the mean. For example, to compute the average number of hours of sleep, apply the `mean` function to the column `:sleep_total` and call the summary value `:avg_sleep`. 
+
+```@example 1
+@chain msleep @combine :avg_sleep = mean(:sleep_total)
+```
+
+There are many other summary statistics you could consider such `std`, `minimum`, `maximum`, `median`, `sum`, `length` (returns the length of vector), `first` (returns first value in vector), and `last` (returns last value in vector).
+
+```@example 1
+@combine msleep begin 
+    avg_sleep = mean(:sleep_total) 
```

Comment:
don't you need `:` prefix in target column names? Maybe also explicitly comment on it?

Reply:
Added a little bit about strings on the LHS.